### ISSUE TITLE
[Character Editor] Crash when adjusting limb sizes on sprite sheet

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#4517]
Branch: dev
Build: Release
Commit: https://github.com/Regalis11/Barotrauma-development/commit/73005a255878af057f861b19347f9364ca9de929

Repro steps: 
1. Select Edit -> Limbs
2. Toggle "Adjust Collider" on
3. Resize a limb so the size adjustable handle goes to negative
4. Crash

![image.png](https://images.zenhubusercontent.com/5c924117534a9b0a86cdddce/c4d3210a-988b-456e-84fa-657d580c1c01)

![image.png](https://images.zenhubusercontent.com/5c924117534a9b0a86cdddce/6643c322-1a49-4101-a6ec-54ffbeb305e5)

```
Barotrauma Client crash report (generated on 14.12.2020 12:02:51)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


BB9C0C7BA78C3A07BE8927D82E8333A3


Game version 0.1200.0.0 (ReleaseWindows, branch dev, revision 73005a2558)
Graphics mode: 2560x1440 (BorderlessWindowed)
VSync OFF
Language: English
Selected content packages: Vanilla 0.9
Level seed: no level loaded
Loaded submarine: AnimEditor (B9856B3F97C56866A95FE75D0038FA16)
Selected screen: Barotrauma.CharacterEditor.CharacterEditorScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.18363.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1,7777778}
    GPU status: Normal


Exception: Texture width must be greater than zero (Parameter 'width') (System.ArgumentOutOfRangeException)
Target site: Void .ctor(Microsoft.Xna.Framework.Graphics.GraphicsDevice, Int32, Int32, Boolean, Microsoft.Xna.Framework.Graphics.SurfaceFormat, SurfaceType, Boolean, Int32)
Stack trace: 
   at Microsoft.Xna.Framework.Graphics.Texture2D..ctor(GraphicsDevice graphicsDevice, Int32 width, Int32 height, Boolean mipmap, SurfaceFormat format, SurfaceType type, Boolean shared, Int32 arraySize) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Graphics\Texture2D.cs:line 102
   at Microsoft.Xna.Framework.Graphics.Texture2D..ctor(GraphicsDevice graphicsDevice, Int32 width, Int32 height) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Graphics\Texture2D.cs:line 48
   at Barotrauma.GUI.<>c__DisplayClass163_0.<CreateCapsule>b__0() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUI.cs:line 1680
   at Barotrauma.CrossThread.RequestExecutionOnMainThread(TaskDelegate deleg) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Utils\CrossThread.cs:line 61
   at Barotrauma.GUI.CreateCapsule(Int32 radius, Int32 height) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GUI\GUI.cs:line 1683
   at Barotrauma.PhysicsBody.DebugDraw(SpriteBatch spriteBatch, Color color, Boolean forceColor) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Physics\PhysicsBody.cs:line 133
   at Barotrauma.CharacterEditor.CharacterEditorScreen.Draw(Double deltaTime, GraphicsDevice graphics, SpriteBatch spriteBatch) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Screens\CharacterEditor\CharacterEditorScreen.cs:line 819
   at Barotrauma.GameMain.Draw(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 1008
   at Microsoft.Xna.Framework.Game.DoDraw(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 669
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 511
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 94
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 398
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 368
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 59


Last debug messages:
[14.12.2020 12:00:01] Content/Characters/Watcher/Watcher.xml
[14.12.2020 12:00:01] Content/Characters/Tigerthresher/Tigerthresher.xml
[14.12.2020 12:00:01] Content/Characters/Terminalcell/Terminalcell.xml
[14.12.2020 12:00:01] Content/Characters/Spineling/Spineling.xml
[14.12.2020 12:00:01] Content/Characters/Psilotoad/Psilotoad.xml
[14.12.2020 12:00:01] Content/Characters/Peanut/Peanut.xml
[14.12.2020 12:00:01] Content/Characters/Orangeboy/Orangeboy.xml
[14.12.2020 12:00:01] Content/Characters/Mudraptor_unarmored/Mudraptor_unarmored.xml
[14.12.2020 12:00:01] Content/Characters/Mudraptor/Mudraptor.xml
[14.12.2020 12:00:01] Content/Characters/Molochblack/Molochblack.xml
[14.12.2020 12:00:01] Content/Characters/Molochbaby/Molochbaby.xml
[14.12.2020 12:00:01] Content/Characters/Moloch/Moloch.xml
[14.12.2020 12:00:01] Content/Characters/Mantis/Mantis.xml
[14.12.2020 12:00:01] Content/Characters/Leucocyte/Leucocyte.xml
[14.12.2020 12:00:01] Content/Characters/Legacywatcher/Legacywatcher.xml
[14.12.2020 12:00:01] Content/Characters/Legacytigerthresher/Legacytigerthresher.xml
[14.12.2020 12:00:01] Content/Characters/Legacymoloch/Legacymoloch.xml
[14.12.2020 12:00:01] Content/Characters/Legacyhusk/Legacyhusk.xml
[14.12.2020 12:00:01] Content/Characters/Legacycrawler/Legacycrawler.xml
[14.12.2020 12:00:01] Content/Characters/Legacycharybdis/Legacycharybdis.xml
[14.12.2020 12:00:01] Content/Characters/Husk/Husk.xml
[14.12.2020 12:00:01] Content/Characters/Humanhusk/Humanhusk.xml
[14.12.2020 12:00:01] Content/Characters/Human/Human.xml
[14.12.2020 12:00:01] Content/Characters/Hammerheadspawn/Hammerheadspawn.xml
[14.12.2020 12:00:01] Content/Characters/Hammerheadmatriarch/Hammerheadmatriarch.xml
[14.12.2020 12:00:01] Content/Characters/Hammerheadgold/Hammerheadgold.xml
[14.12.2020 12:00:01] Content/Characters/Hammerhead/Hammerhead.xml
[14.12.2020 12:00:01] Content/Characters/Fractalguardian2/Fractalguardian2.xml
[14.12.2020 12:00:01] Content/Characters/Fractalguardian/Fractalguardian.xml
[14.12.2020 12:00:01] Content/Characters/Endworm/Endworm.xml
[14.12.2020 12:00:01] Content/Characters/Crawlerhusk/Crawlerhusk.xml
[14.12.2020 12:00:01] Content/Characters/Crawler/Crawler.xml
[14.12.2020 12:00:01] Content/Characters/Coelanth/Coelanth.xml
[14.12.2020 12:00:01] Content/Characters/Charybdis/Charybdis.xml
[14.12.2020 12:00:01] Content/Characters/Carrier/Carrier.xml
[14.12.2020 12:00:01] Content/Characters/Bonethresher/Bonethresher.xml
[14.12.2020 12:00:01] Content/Characters/Balloon/Balloon.xml
[14.12.2020 12:00:01] Trying to spawn Content/Characters/Human/Human.xml
[14.12.2020 11:59:49] (Msg) SDR RelayNetworkStatus:  avail=OK  config=OK  anyrelay=OK   (OK)
[14.12.2020 11:59:49] (Msg) Ping location: sto=15+1,sto2=15+1,ams=38+3/35+1,fra=39+3/35+1,waw=44+4/35+1,lux=46+4/39+1,lhr=45+4/40+1,par=51+5/44+1,vie=51+5/47+1,iad=128+12/120+1,sgp=210+21/211+21,gru=247+24/264+1
[14.12.2020 11:59:49] (Msg) Ping measurement completed
[14.12.2020 11:59:47] (Debug) Destroying relay 'sof#2 (139.45.193.10:27016)' because initial_ping_timeout
[14.12.2020 11:59:46] WARNING: Cannot compress a texture because the dimensions are not a multiple of 4 (path: Content/Particles/vomitdecal.png, size: 126x126)
[14.12.2020 11:59:45] (Debug) Destroying relay 'sof#5 (139.45.193.10:27019)' because initial_ping_timeout
[14.12.2020 11:59:42] (Debug) Destroying relay 'sof#1 (139.45.193.10:27015)' because initial_ping_timeout
[14.12.2020 11:59:40] (Debug) Destroying relay 'sof#3 (139.45.193.10:27017)' because initial_ping_timeout
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'syd' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'tyo1' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'tyo' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'sgp' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'sea' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'scl' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'lim' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'maa' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'hkg' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'lax' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'okc' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'vie' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'waw' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'gru' established.
[14.12.2020 11:59:37] (Msg) Relay ord#102 (162.254.193.73:27032) is going offline in 282 seconds
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'ord' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'jnb' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'sto' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'sto2' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'eat' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'bom' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'mad' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'par' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'dxb' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'lux' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'lhr' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'atl' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'iad' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'fra' established.
[14.12.2020 11:59:37] (Verbose) Communications with routing cluster 'ams' established.
[14.12.2020 11:59:37] (Msg) SDR RelayNetworkStatus:  avail=Attempting  config=OK  anyrelay=Attempting   (Performing ping measurement)
[14.12.2020 11:59:37] (Msg) Performing ping measurement
[14.12.2020 11:59:37] (Msg) Got SDR network config.  Loaded revision 286 OK
[14.12.2020 11:59:37] (Warning) SDR network config fetch first attempt failed.  HTTP 504.  .  Trying again.
[14.12.2020 11:59:30] Attempting to open ALC device "OpenAL Soft on Kaiuttimet (Realtek High Definition Audio)"
[14.12.2020 11:59:29] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
[14.12.2020 11:59:29] Loaded languages: English, German, French, Russian, Polish, Brazilian Portuguese, Castilian Spanish, Latinamerican Spanish, Simplified Chinese, Traditional Chinese, Japanese, Turkish, Korean
```

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@f3ef40b

**Comment by Rokvach** [comment]
Tested, wasn't able to reproduce. Working correctly. Closing.
